### PR TITLE
Initial Contrib Http Instrumentation Components

### PR DESCRIPTION
**Pull request by kbrockhoff**
Creating this for visibility. Covers JEE Servlet and JAX-RS specifications with metrics mocked out until MeterSdk implementation is complete.
Partially completes #522

### CODE REVIEW

**Comment by bogdandrutu** [comment]
Hi @kbrockhoff,

First thanks a lot for your contributions. I would encourage to split this PR into smaller more reviewable PRs so that we can review the code.

**Comment by jkwatson** [comment]
Yes, to re-iterate @bogdandrutu's comment...let's work on breaking this up into smaller, digestible chunks. I can probably start helping out with that in a week or so, if you'd like some assistance.

**Comment by kbrockhoff** [comment]
I can break it up into per module chunks. I was holding off until MeterSdk is implemented so functionality is complete. At that point I plan to remove the stubbed out metrics classes and adding the real measurement recording code. If the group would prefer to merge now and then do another set of pull requests later for metrics, let me know. I will get it done.

File: contrib/http_base/src/main/java/io/opentelemetry/contrib/http/core/HttpStatus2OtStatusConverter.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.http.core;
+
+import io.opentelemetry.trace.Status;
+import javax.annotation.Nullable;
+
+/** Provides standard translation from HTTP status codes to Open Telemetry trace status values. */
+public class HttpStatus2OtStatusConverter {
```
Comment (jkwatson) [inline contrib/http_base/src/main/java/io/opentelemetry/contrib/http/core/HttpStatus2OtStatusConverter.java]: this name is 👎  How about just `StatusCodeConverter`

File: contrib/http_base/src/main/java/io/opentelemetry/contrib/http/core/HttpStatus2OtStatusConverter.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  public Status convert(int httpStatus) {
+    Status status = doCustomConvert(httpStatus);
+    if (status != null) {
+      return status;
+    }
+    if (httpStatus >= 200 && httpStatus < 400) {
+      status = Status.OK;
+    } else if (httpStatus == 401) {
+      status = Status.UNAUTHENTICATED;
```
Comment (jkwatson) [inline contrib/http_base/src/main/java/io/opentelemetry/contrib/http/core/HttpStatus2OtStatusConverter.java]: what would you think about putting these "special case" codes into a Map, and looking them up first before the range-based logic?

**Comment by jkwatson** [comment]
When you rebase against master, this will end up with a lot of updates needed, due to all the renaming/cleanup of the Metric API, just FYI.

File: contrib/http_servlet/src/main/java/io/opentelemetry/contrib/http/servlet/OtelHttpServletUtils.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.contrib.http.core.HttpRequestContext;
+import io.opentelemetry.contrib.http.core.HttpServerHandler;
+import io.opentelemetry.trace.DefaultSpan;
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Provides static utility methods for working with Java Servlet objects. */
+public class OtelHttpServletUtils {
```
Comment (jkwatson) [inline contrib/http_servlet/src/main/java/io/opentelemetry/contrib/http/servlet/OtelHttpServletUtils.java]: I feel like having the `Otel` prefix is everywhere unnecessary. This is all OpenTelemetry code. :)

**Comment by kbrockhoff** [comment]
Closing and resubmitting in smaller chunks

**Pull request closed without merging**